File: 283-MoveZeroes/283-MoveZeroes.py
# Last updated: 4/29/2026, 10:31:49 AM
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        readIdx = 0
        writeIdx = 0
        
        while readIdx < len(nums):
            num = nums[readIdx]
            if num != 0:
                nums[writeIdx] = num
                writeIdx += 1
            readIdx += 1
        
        while writeIdx < len(nums):
            nums[writeIdx] = 0
            writeIdx += 1
        

File: 283-MoveZeroes/283-MoveZeroes.js
// Last updated: 4/29/2026, 10:31:45 AM
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var moveZeroes = function(nums) {
    let writeIdx = 0;
    const n = nums.length;
    for(let i = 0; i < n; i++) {
        console.log(`i: ${i}, writeIdx: ${writeIdx}`)
        const num = nums[i];
        if(num !== 0) {
            nums[writeIdx] = num;
            writeIdx++;
        }
    }
    for(let j = writeIdx; j < n; j++) {
        nums[j] = 0;
    }
};